Failed to compile Cython code in CI for Python 3.6

As Windows image of Github Actions now contains VS2015 and Python 3.6 choose it as default compiler, we encounter failure when compiling Cython code.

```
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\x86_amd64\link.exe /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFESTUAC:NO /LIBPATH:c:\miniconda\envs\test\libs /LIBPATH:c:\miniconda\envs\test\PCbuild\amd64 "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\LIB\amd64" "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\LIB\amd64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.18362.0\ucrt\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.18362.0\um\x64" /EXPORT:PyInit_graph build\temp.win-amd64-3.6\Release\mars/graph.obj /OUT:D:\a\mars\mars\mars\graph.cp36-win_amd64.pyd /IMPLIB:build\temp.win-amd64-3.6\Release\mars\graph.cp36-win_amd64.lib
    graph.obj : warning LNK4197: export 'PyInit_graph' specified multiple times; using first specification
       Creating library build\temp.win-amd64-3.6\Release\mars\graph.cp36-win_amd64.lib and object build\temp.win-amd64-3.6\Release\mars\graph.cp36-win_amd64.exp
    Generating code
    Finished generating code
    LINK : fatal error LNK1158: cannot run 'rc.exe'
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\x86_amd64\\link.exe' failed with exit status 1158
```

--- a/.github/workflows/install-conda.sh
+++ b/.github/workflows/install-conda.sh
@@ -13,12 +13,14 @@ elif [[ $UNAME == "mingw"* ]] || [[ $UNAME == "msys"* ]]; then
   CONDA_OS="Windows"
   FILE_EXT="exe"
   # VS2015 needed by Python 3.5
-  if [[ "$PYTHON" == "3.5" ]]; then
-    npm install --global --silent --vs2015 windows-build-tools
+  if [[ "$PYTHON" == "3.5" ]] || [[ "$PYTHON" == "3.6" ]]; then
+    vcbindir="/c/Program Files (x86)/Microsoft Visual Studio 14.0/VC/BIN/x86_amd64"
+    if [[ ! -d $vcbindir ]]; then
+      npm install --global --silent --vs2015 windows-build-tools
+    fi
     # Remove SDK of Windows Driver Framework to avoid confusion
     rm -rf "/c/Program Files (x86)/Windows Kits/10/include/wdf"
     # Locate and copy resource compiler into VC dir
-    vcbindir="/c/Program Files (x86)/Microsoft Visual Studio 14.0/VC/BIN/x86_amd64"
     for subdir in $(ls -r "/c/Program Files (x86)/Windows Kits/10/bin"); do
       bindir="/c/Program Files (x86)/Windows Kits/10/bin/$subdir/x64"
       if [ -f "$bindir/rc.exe" ] && [ -f "$bindir/rcdll.dll" ]; then

--- a/requirements-extra.txt
+++ b/requirements-extra.txt
@@ -4,6 +4,6 @@ numexpr>=2.6.4
 psutil>=4.0.0
 lz4>=1.0.0
 protobuf>=3.6
-gevent>=1.3.0
+gevent>=1.2.0
 bokeh>=1.0.0
 jinja2>=2.0

